### PR TITLE
Fix exception messages

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Extends | https://github.com/yiisoft/di/pull/315
| Depends on | https://github.com/yiisoft/definitions/pull/56

### CODE REVIEW

File: tests/Unit/ContainerTest.php [hunk 1]
```diff
@@ -1580,7 +1584,10 @@ public function getExtensions(): array
             }
         };
 
-        $this->expectException(RuntimeException::class);
+        $this->expectException(BuildingException::class);
+        $this->expectExceptionMessage(
+            'Caught unhandled error "" while building "Yiisoft\Di\Tests\Support\B".'
```
Comment (samdark) [inline tests/Unit/ContainerTest.php:1589]: Empty error?

Comment (xepozz) [reply tests/Unit/ContainerTest.php:1589]: RuntimeException doesn't provide any messages

Comment (samdark) [reply tests/Unit/ContainerTest.php:1589]: Why do we need `""` then in the message?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Tests seems to fail: https://github.com/yiisoft/di/actions/runs/3604857565/jobs/6074657047

**Pull request merged**